lib/BaseDownloadController.js

```diff
@@ -19,9 +19,9 @@ const yaml = require('js-yaml');
 const fs = require('fs-extra');
 const hash = require('object-hash');
 const clone = require('clone');
-
+const { request } = require('@octokit/request');
+const gitFactory = require('./git');
```

Comment:
the git factory should live in RemoteResource not razeedeploy-core, and it should be extending the baseDownloadController class and defining the download() function for all your specialized logic.

- [Extend the base clase example](https://github.com/razee-io/RemoteResource/blob/master/src/RemoteResourceS3Controller.js#L31)
- [define the specialized download logic example](https://github.com/razee-io/RemoteResource/blob/master/src/RemoteResourceS3Controller.js#L57-L76)
- I dont think you need to do this, but in case you do.. the s3 factory had to do some pre-processing on the request urls. so it also overloaded the `added()` function. [ref](https://github.com/razee-io/RemoteResource/blob/master/src/RemoteResourceS3Controller.js#L39-L55)

for context - at a high level, the way this works is like this:
1. a new resource comes in and the BaseDownloadController [`added()` function](https://github.com/razee-io/razeedeploy-core/blob/master/lib/BaseDownloadController.js#L33) is triggered.
2. within the baseDownloadController `added()` function, the rest of the download flow gets triggered:
    1. [manage file caching](https://github.com/razee-io/razeedeploy-core/blob/master/lib/BaseDownloadController.js#L36-L63) - use local data cache unless the file has been modified
    2. [fetching secret header data](https://github.com/razee-io/razeedeploy-core/blob/master/lib/BaseDownloadController.js#L66) - if any of the headers of secret refs it will fetch the data
    3. [download the file](https://github.com/razee-io/razeedeploy-core/blob/master/lib/BaseDownloadController.js#L81) - this is the piece that actually [isnt defined](https://github.com/razee-io/razeedeploy-core/blob/master/lib/BaseDownloadController.js#L247-L252) in baseDownloadController, but in [remoteresource factories instead](https://github.com/razee-io/RemoteResource/blob/master/src/RemoteResourceS3Controller.js#L57). baseDownloadController expects the class that extends it (ie [RemoteResourceS3Controller](https://github.com/razee-io/RemoteResource/blob/master/src/RemoteResourceS3Controller.js)) will define the `download()` function which has the specialized logic on how to do that download.
    4. finally, it [applies the file](https://github.com/razee-io/razeedeploy-core/blob/master/lib/BaseDownloadController.js#L119) that got downloaded to the cluster